c_binding/lsm_mgmt.cpp

```diff
@@ -2610,3 +2613,43 @@ int lsm_system_read_cache_pct_update(lsm_connect * c, lsm_system * system,
 
     return rc;
 }
+
+static int get_battery_array(lsm_connect *c, int rc, Value &response,
+                             lsm_battery **bs[], uint32_t *count)
+{
+    if (LSM_ERR_OK == rc && Value::array_t == response.valueType()) {
+        rc = value_array_to_batteries(response, bs, count);
```

Comment:
We need to wrap the call to `value_array_to_batteries` with a `try.. catch` see  `get_volume_array`.  This is because `value_to_battery` can raise an exception that must be caught.